### ISSUE TITLE
Move to 'faker' on Pypi

### ISSUE THREAD

**Issue by jeffwidman** [joke2k/faker#331]
Looks like the `Faker` namespace on Pypi has been abandoned for quite a while: https://pypi.python.org/pypi/Faker

This project's owner @joke2k should file a request to have the namespace transferred to this project. 

It would reduce confusion between Faker (project name) and `fake-factory` (how it gets pip installed). 

The pypi team is pretty good about transferring namespaces if they're clearly abandoned (which seems to be the case here).

**Comment by kdeldycke** [comment]
Relates to #47.

**Comment by fcurella** [comment]
it's worth noticing that while the last upload on PyPI was in 2010, the GH repo had its last commit on Aug 6th, 2015 - which I would  not consider *that* long ago.

**Comment by fcurella** [comment]
Alternatively, `python-faker` hasn't been updated on PyPI since since 2011 (https://pypi.python.org/pypi/python-faker/0.2.4) and the link to the repo is a 404, so it might be easier to claim.

**Comment by jeffwidman** [comment]
I'm cool with whatever, I just think from a branding perspective, it'd be best to have the same name across Pypi/GitHub, also grab the handle on RTD just to avoid someone else grabbing it and adding confusion etc. Plus it'd be nice if it was consistent when you do `import Faker` or whatever to make sure the package/class name is also consistent. 

I'm personally not too bothered by having a unique name outside of the python ecosystem as suggested in #47. I think being unique within the python ecosystem is what matters.

**Comment by deepthawtz** [comment]
I'm the author of the "faker" package on PyPI. I am a crappy OSS maintainer and I can barely code Python anymore :stuck_out_tongue: I don't see the point keeping that project going and am happy to point to this package and/or give up the "faker" PyPi namespace... let me know what I can do to help

**Comment by deepthawtz** [comment]
Yesterday I got around to update the "faker" PyPi package but since I've been out of the Python loop for so long had no idea about this project until I googled around

**Comment by fcurella** [comment]
Hi @deepthawtz , thanks for getting in touch!

> I don't see the point keeping that project going and am happy to point to this package and/or give up the "faker" PyPi namespace... let me know what I can do to help

You can add 'joke2k' as owner to the package, and 'curella.org' (that's my username for historical reasons) as 'maintainer'.

As for the future of `fake-factory`, I need to check exactly how removing a package works.

If the name gets released, I don't want anyone to squat the name and potentially distribute malicious code under that name. In that case, I think we can upload a new version that simply raises an helpful error. We may have a 'grace period' (i'm thinking 6 months) during which we upload to both and issue a warning instead of raising the error.

Any thoughts on how to handle the transition?

**Comment by jeffwidman** [comment]
@deepthawtz Thanks!

@fcurella I dont have answers, but I do know the `pep8` package [is in the midst of a Pypi name change](https://github.com/PyCQA/pycodestyle/issues/466) so may be instructive to watch how they handle it.

**Comment by deepthawtz** [comment]
@fcurella I pushed a new Faker release to PyPi with deprecation warning announcement and set the date for September 15, 2016 to give folks time to update. If you can mark your calendar too that'd be great. I'm now instructing everyone to use your package instead. I've marked my calendar and we'll be in touch in 6 months. I'm going to hold off transferring ownership until we actually make the package switch.

**Comment by fcurella** [comment]
thanks @deepthawtz .

I think this is a good, but we will also need some period of time during which both names have the same package, with the warning only on the `fake-factory` side. Maybe that can start 3 months from now (ie: June 15th)?

**Comment by deepthawtz** [comment]
Are you proposing I update the warning message and make the transition period 3 months instead?

**Comment by fcurella** [comment]
No, I think what you've done is perfect so far.

What I'm proposing is that in 3 month we will upload the code in this repo to _both_ 'fake-factory` and `factory`, with just two small differences:

1. The `faker` package will not have any warning
2. The `fake-factory` package will have a warning

Then, on Sept. 15h, we will pull the plug on `fake-factory`. We either replace it with an 'empty' package (ie: it will contain _just_ the warning) or we remove the package completely from PyPI.

Basically the timeline is still 6 months, but within it we have 3 months of 'overlap' to help transition.

**Comment by fcurella** [comment]
for future reference, something like this could be helpful https://github.com/pytest-dev/pytest/blob/1f6988bdec07caa0c6488a488b390beeed32e529/extra/setup-py.test/setup.py

**Comment by deepthawtz** [comment]
if the PyPi faker package changes to your package in 3 months that is different than what I've communicated in the deprecation message. I indicated that after Sept 15 the package will be your package.

```
This faker package is being deprecated September 15, 2016.
You should switch to using https://pypi.python.org/pypi/fake-factory instead.
After September 15, 2016 the PyPi faker package will be changing to that!
```

Wondering if I should communicate that differently based on your proposal.

**Comment by fcurella** [comment]
@deepthawtz yes, you're correct. According to my proposal, that date should say June 15th.

You can keep the message as it is, and we can move the timeline by 3 months. Basically:

* Sept. 15th: Start transition period. This repo gets uploaded to both, warning only on `fake-factory`
* Dec. 15th: End transition period. `fake-factory` gets replaced by a 'shell' package.

**Comment by deepthawtz** [comment]
:+1: works for me, glad we could work together on this

**Comment by fcurella** [comment]
@jeffwidman thanks for pointing out to the `pep8` case :)

Thankfully our situation is a lot simpler, since we don't need to rename the repo or the module name.

Nonetheless it's a really useful thread, we'll definitely steal something of that they do :)

**Comment by fcurella** [comment]
Note for future self:

- [ ] submit PR with updated requirements to FactoryBoy
- [ ] submit PR with updated reqs to Hypothesis

**Comment by jeffwidman** [comment]
@fcurella I'm a maintainer on FactoryBoy and happy to merge this. 

Long term, we've got a goal to get rid of some our random code stuff in favor of faker plugins.

**Comment by folz** [comment]
Happy September 15, 2016 :) Didn't notice a warning when installing/using fake-factory so figured I'd bump this thread.

**Comment by fcurella** [comment]
@deepthawtz I need you to give me maintainer access to the PyPI package. Can you add `curella.org` under 'roles' ?

**Comment by deepthawtz** [comment]
done, thank you

**Comment by fcurella** [comment]
Thanks @deepthawtz 

I've uploaded`Faker` on PyPI. To help with telling them apart, I'm keeping them at slight different version, with `Faker` always one patch version number higher than `fake-factory`.

* `Faker` is currently `v0.7.3`
* `fake-factory` is at `v0.7.2`

The only difference between the two codebase is the package name in `setup.py`.

**Comment by jeffwidman** [comment]
Is there a plan to eventually no longer update the `fake-factory` namespace? It just creates extra work for the maintainers long term... maybe make it so the final release of `fake-factory` throws a runtime warning saying that the package location has moved...

**Comment by fcurella** [comment]
@jeffwidman  The plan is to replace `fake-factory` with an 'empty' package that just prints out a message pointing to `Faker` on December 15th. I have no desire to keep two slighlty different branches and packages in sync for longer than 90 days

**Comment by folz** [comment]
It might be a good idea to start throwing a quiet warning now (like
Django's PendingDeprecationWarninan and switch to a loud warning a month or
two before (DeprecationWarning) so that people who don't actively seek out
GH issues will be aware this change is happening. December 15 is near
Christmas break and I'd hate to see this namechange keep Ops folks from
their vacation.

**Comment by fcurella** [comment]
@folz I've tried `warnings.warn()` but it wouldn't print out the message when installing with `pip`, so I had to resort to `print()`

The Dec 15th deadline was decided so to give people 90 days to notice and switch. I don't want to keep maintaining both packages any longer, but I also want to give people enough time. Dec 15th is ~9 days from Christmas break, I'd think it's  far enough.

**Comment by fcurella** [comment]
@deepthawtz could you grant me or @joke2k the 'owner' role on PyPI?

**Comment by deepthawtz** [comment]
done

**Comment by fcurella** [comment]
Thanks!

I've removed @deepthawtz from the roles, added @joke2k as owner, and
reverted myself to maintainer.
